x/slashing/keeper.go

```diff
@@ -120,3 +126,50 @@ func (k Keeper) handleValidatorSignature(ctx sdk.Context, pubkey crypto.PubKey,
 	// Set the updated signing info
 	k.setValidatorSigningInfo(ctx, address, signInfo)
 }
+
+// AddValidators adds the validators to the keepers validator addr to pubkey mapping.
+func (k Keeper) AddValidators(ctx sdk.Context, vals []abci.Validator) {
+	for i := 0; i < len(vals); i++ {
+		val := vals[i]
+		pubkey, err := tmtypes.PB2TM.PubKey(val.PubKey)
+		if err != nil {
```

Comment:
Should we panic here?

Reply:
Why wouldn't we want to panic here? We currently only call this in EndBlock, and presumably we shouldn't be calling it with invalid Validators